dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapConnection.java

```diff
@@ -2114,27 +2122,39 @@ private void supportRangeAttributes(Attributes attributes, String dn, TimedDirCo
                         }
                     }
                     Attribute result = results.get(attributeWithRange);
+                    //If the result is null, we didn't find the exact attributeWithRange string
+                    //we were looking for. This could mean there are fewer results left than iAttrRangeStep.
+                    //In this case, LDAP will have sent back range;x-*.
                     if (result == null) {
-                        Object[] args = {
-                                          Integer.valueOf(rangeLow).toString()
+                        Object[] rangeArgs = {
+                                               new Integer(rangeLow).toString()
```

Comment:
This would be more efficient:

```suggestion
                                               Integer.toString(rangeLow)
```